osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -94,6 +94,12 @@ public virtual string SettingDescription
         [JsonIgnore]
         public virtual bool UserPlayable => true;
 
+        [JsonIgnore]
+        public virtual bool PlayableInMultiplayer => UserPlayable;
+
+        [JsonIgnore]
+        public virtual bool ValidFreeModInMultiplayer => PlayableInMultiplayer;
```

Comment:
These cascading fallbacks make me scared. How am I supposed to tell which value `ValidFreeModInMultiplayer`? Now I have to chase 2 other virtual members by default just to make sure.

Would prefer these to be all explicit and not cascading-order-magic.

Reply:
A follow up thought is that with three properties this is likely going to be confusing anyways. Things and places are going to forget to check this.

Suggestion would be to centralise checking mod validity to a single source-of-truth method as follows:

```csharp
public enum ModUsageContext
{
    Solo,
    MultiplayerRequiredMod,
    MultiplayerFreeMod
}

public

Reply:
While this could work in some cases, there are other cases where the caller is shared between solo and multiplayer, e.g.:

https://github.com/ppy/osu/blob/6bf436cd62d0c844d74964837fb2387ebe112725/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs#L190-L191

In which case we want to ensure the mod is playable at a per-user level, which `Solo` doesn't convey it well. How about calling it `User` instead (and add a xmldoc to it).

Also it slightly feels weird that you could potentially have a mod that has `IsPlayable(ModUsage.MultiplayerRequired) = false` but `IsPlayable(ModUsage.MultiplayerFree) = false`. Would you still be against renaming `MultiplayerRequiredMod` to become just `Multiplayer` in general, and cover `MultiplayerFreeMod` as a result.

Reply:
I don't really like `User` as a choice here and the mentioned check could just use `Solo` because the user's own score is what actually matters in the context of that particular call.

Reply:
Hmm maybe, let's leave it at `Solo` then.

Reply:
I've pushed the changes so you can look at it and see if it's good as-is.